src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -360,14 +363,37 @@ public Rotation2d getYawPerSecond() {
 
         @Override
         public double getLinearSpeed() {
-                velocityXMPS = getRobotRelativeSpeeds().vxMetersPerSecond;
-                velocityYMPS = getRobotRelativeSpeeds().vyMetersPerSecond;
-                return Math.sqrt((Math.pow(velocityXMPS, 2) + Math.pow(velocityYMPS, 2)));
+                double velocityXMPS = getRobotRelativeSpeeds().vxMetersPerSecond;
+                double velocityYMPS = getRobotRelativeSpeeds().vyMetersPerSecond;
+                return Math.hypot(velocityXMPS, velocityYMPS);
         }
 
         @Override
         public void addVisionMeasurement(Pose2d pose, double timestampSeconds,
                         Matrix<N3, N1> visionMeasurementStdDevs) {
                 poseEstimator.addVisionMeasurement(pose, timestampSeconds, visionMeasurementStdDevs);
         }
-}
\ No newline at end of file
+
+        private void logAndUpdateDriveSubsystemStates() {
+                states.pose = getPose();
+                states.poseX = states.pose.getX();
+                states.poseY = states.pose.getY();
```

Comment:
Should be able to remove these two since the values for logging can be gotten using states.pose